File: src/test/java/net/sf/rails/game/state/StateImpl.java
package net.sf.rails.game.state;

import net.sf.rails.game.state.Item;
import net.sf.rails.game.state.State;

// An implementation only for testing
class StateImpl extends State {

    private final String text;
    
    private StateImpl(Item parent, String id, String text) {
        super(parent, id);
        this.text = text;
    }
    
    static StateImpl create(Item parent, String id, String text) {
        return new StateImpl(parent, id, text);
    }
    
    @Override
    public String toText() {
        return text;
    }
    
    @Override
    public String toString() {
        return text;
    }
}

File: dotnet/GameStateTest/StateImpl.cs
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GameStateTest
{
    public class StateImpl : GameState
    {
        private string text;
    
    private StateImpl(IItem parent, string id, string text) : base(parent, id)
        {
            this.text = text;
        }

        public static StateImpl Create(IItem parent, string id, string text)
        {
            return new StateImpl(parent, id, text);
        }

    override public string ToText()
        {
            return text;
        }

    override public string ToString()
        {
            return text;
        }
    }
}
